### PR TITLE
Celery

### PR DESCRIPTION
**Pull request by xingyifei2016**
Implement celery

### CODE REVIEW

File: clusterCAD/celery.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from __future__ import absolute_import, unicode_literals
+import os
+from celery import Celery
+from django.conf import settings
+
+# import sequenceSearch
```
Comment (TylerBackman) [inline clusterCAD/celery.py]: delete this

File: clusterCAD/celery.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# Using a string here means the worker doesn't have to serialize
+# the configuration object to child processes.
+# - namespace='CELERY' means all celery-related configuration keys
+#   should have a `CELERY_` prefix.
+app.config_from_object('django.conf:settings', namespace='CELERY')
+
+# Load task modules from all registered Django app configs.
+app.autodiscover_tasks()
+
+# @app.task
```
Comment (TylerBackman) [inline clusterCAD/celery.py]: Describe this debug comment block

File: clusterCAD/celery.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import os
+from celery import Celery
+from django.conf import settings
+
+# import sequenceSearch
+
+# set the default Django settings module for the 'celery' program.
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'clusterCAD.settings')
+
+app = Celery('sequenceSearch')
```
Comment (TylerBackman) [inline clusterCAD/celery.py]: make this clusterCAD

File: clusterCAD/example_settings.py [hunk 4]
```diff
@@ -12,6 +12,21 @@
 
 import os
 
+
+# CELERY_TASK_SERIALIZER = 'pickle'
```
Comment (TylerBackman) [inline clusterCAD/example_settings.py]: delete commented out lines

File: clusterCAD/example_settings.py [hunk 5]
```diff
@@ -12,6 +12,21 @@
...
+# CELERY_ACCEPT_CONTENT = {'pickle'}
+CELERY_BROKER_URL= 'amqp://admin:mypass@rabbit:5672/clusterCAD' # "amqp://guest@localhost//"
+CELERY_RESULT_BACKEND="rpc://"
+CELERY_TASK_ACKS_LATE = True
+CELERY_WORKER_PREFETCH_MULTIPLIER = 1
+CELERYD_CHDIR="/www/clusterCAD"
+CELERYD_TASK_TIME_LIMIT=1
+CELERYD_TASK_SOFT_TIME_LIMIT=1
+
+gettext = lambda s: s
```
Comment (TylerBackman) [inline clusterCAD/example_settings.py]: drop this unused function, and remove the whitespace lines around it

File: compounddb/models.py [hunk 6]
```diff
@@ -3,6 +3,7 @@
 from django.db import connection
 from django.db.models.signals import pre_delete
 from django.dispatch import receiver
+from django.conf import settings
```
Comment (TylerBackman) [inline compounddb/models.py]: remove

File: debian-cheminformatics/Dockerfile [hunk 7]
```diff
@@ -9,6 +9,7 @@ ENV LANG C.UTF-8
 WORKDIR /root
 ADD ./requirements.txt /root/
 
+
```
Comment (TylerBackman) [inline debian-cheminformatics/Dockerfile]: remove

File: debian-cheminformatics/Dockerfile [hunk 8]
```diff
@@ -56,6 +57,7 @@ RUN apt-get update \
 RUN pip3 install -U pip setuptools
 RUN pip3 install -r requirements.txt
 
+
```
Comment (TylerBackman) [inline debian-cheminformatics/Dockerfile]: remove

File: pks/migrations/0002_auto_20191009_0546.py [hunk 9]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 2.2.3 on 2019-10-09 05:46
+
+from django.db import migrations, models
```
Comment (TylerBackman) [inline pks/migrations/0002_auto_20191009_0546.py]: See if this is needed, if not delete the entire file

File: pks/models.py [hunk 10]
```diff
@@ -1,4 +1,5 @@
 import sys
+from django.conf import settings
```
Comment (TylerBackman) [inline pks/models.py]: remove

File: sequenceSearch/celery.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
+from __future__ import absolute_import, unicode_literals
```
Comment (TylerBackman) [inline sequenceSearch/celery.py]: Remove this file and confirm if it works- let's use the global celery app for all of clustercad

File: sequenceSearch/sequencetools.py [hunk 12]
```diff
@@ -45,48 +48,51 @@ def blast(
...
     blastp_cline = NcbiblastpCommandline(
                                          db=database,
                                          evalue=evalue,
-                                         outfmt=5,
+                                         outfmt=7,
                                          num_threads=2
                                          )
     result, stderr = blastp_cline(stdin=queryFasta)
 
+    queries_found = True
```
Comment (TylerBackman) [inline sequenceSearch/sequencetools.py]: remove

File: sequenceSearch/sequencetools.py [hunk 13]
```diff
@@ -45,48 +48,51 @@ def blast(
...
+
+    results = result.split("\n")
+    columns = results[3].split(":")[-1].split(",")
+    columns = [i.lstrip() for i in columns]
+
+    if results[3] == "# 0 hits found":
+        return None, None, False
+
+    # assert False, result
```
Comment (TylerBackman) [inline sequenceSearch/sequencetools.py]: remove

File: sequenceSearch/sequencetools.py [hunk 14]
```diff
@@ -45,48 +48,51 @@ def blast(
...
     # parse blast output and delete files
     resultIO = StringIO(result)
-    blast_record = NCBIXML.read(resultIO)
+
+    #Remeber bitscore cant convert to int by rule 'safe'
+    # df = pd.read_csv(resultIO, sep="\s+", comment="#", names=columns, dtype={'bit score': 'int32'})
```
Comment (TylerBackman) [inline sequenceSearch/sequencetools.py]: remove

File: sequenceSearch/sequencetools.py [hunk 15]
```diff
@@ -45,48 +48,51 @@ def blast(
...
-        alignments = sorted(individualHSPs, key=lambda alignment: alignment['hsps'][0]['hsp'].bits, reverse=True)[0:max_target_seqs]
-
-    return alignments, str(int(end-start))
+        df = df.sort_values('bit score', ascending=False)
+
+    if max_target_seqs < len(df):
+        df = df[:max_target_seqs]
+
+    # #Get the mibigAccession and subunitID for each query
+    # df[['mibigAccession','subunitID']] = df["subject acc.ver"].str.split('_',expand=True)
```
Comment (TylerBackman) [inline sequenceSearch/sequencetools.py]: remove

File: sequenceSearch/tasks.py [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'clusterCAD.settings')
+
+sys.path.append('../')
+
+
+from clusterCAD.celery import app
+
+
+@app.task()
+def blast(query,
```
Comment (TylerBackman) [inline sequenceSearch/tasks.py]: look into passing all args directly without specifying them

**Comment by xingyifei2016** [comment]
I have made changes, please review

**Pull request merged**